src/hooks/use-measure-text/index.ts

```diff
@@ -0,0 +1,111 @@
+import { memoize } from 'qlik-chart-modules';
+import { useMemo } from 'react';
+
+type EstimateLineCountProps = {
+  text: string;
+  maxWidth: number;
+  isNumeric?: boolean;
+};
+
+export type EstimateLineCount = ({ text, maxWidth, isNumeric }: EstimateLineCountProps) => number;
+
+export interface MeasureTextHook {
+  estimateWidth: (length: number) => number;
+  measureText: (text: string) => number;
+  estimateLineCount: EstimateLineCount;
+}
+
+export interface UseMeasureTextProps {
+  fontSize?: string;
+  fontFamily?: string;
+  fontStyle?: string;
+  bold?: boolean;
+  maxNbrLinesOfText?: number;
+}
+
+const MAX_NBR_LINES_OF_TEXT = 3;
+
+const MAGIC_DEFAULT_CHAR = 'N';
+
+function getNextLine(text: string, maxWidth: number, fixedMeasureText: (text: string) => number) {
+  let left = 0;
+  let right = text.length;
+
+  if (text.length <= 1) {
+    return null;
+  }
+
+  if (fixedMeasureText(text) <= maxWidth) {
+    return null;
+  }
+
+  while (left <= right) {
+    const m = Math.floor((left + right) / 2);
+    const chunk = text.slice(0, m);
+    const prevChunk = text.slice(0, m - 1);
+    const width = fixedMeasureText(chunk);
+    const prevWidth = fixedMeasureText(prevChunk);
+
+    if (width > maxWidth && prevWidth <= maxWidth) {
+      return text.slice(m - 1) || null;
+    }
+
+    if (width === maxWidth) {
+      return text.slice(m) || null;
+    }
+
+    if (width < maxWidth) {
+      // Search right
+      left = m + 1;
+    } else {
+      // Search left
+      right = m;
+    }
+  }
+
+  return null;
+}
+
+export default function useMeasureText({
+  fontSize = '12px',
+  fontFamily = 'Source Sans Pro, Arial, sans-serif',
+  fontStyle = '',
+  bold = false,
+  maxNbrLinesOfText = MAX_NBR_LINES_OF_TEXT,
+}: UseMeasureTextProps): MeasureTextHook {
+  const { estimateWidth, measureText, estimateLineCount } = useMemo((): MeasureTextHook => {
```

Comment:
```suggestion
  const { estimateWidth, measureText, estimateLineCount } = useMemo<MeasureTextHook>(() => {
```

`useMemo<>(() => {...})` accepts a generic type that indicates it's return value